Use vmd's fixed PCI apertures and do not program nonexistent QEMU PIIX/Q35
power-management PCI registers.

Index: edk2/OvmfPkg/Library/PlatformInitLib/Platform.c
--- edk2/OvmfPkg/Library/PlatformInitLib/Platform.c.orig
+++ edk2/OvmfPkg/Library/PlatformInitLib/Platform.c
@@ -171,9 +171,15 @@ PlatformMemMapInitialization (
   // 0xFEE00000    LAPIC                          1 MB
   //
   PlatformGetSystemMemorySizeBelow4gb (PlatformInfoHob);
-  PciBase      = PlatformInfoHob->Uc32Base;
   PciExBarBase = 0;
-  if (PlatformInfoHob->HostBridgeDevId == INTEL_Q35_MCH_DEVICE_ID) {
+  if (PlatformInfoHob->HostBridgeDevId == VMD_HOSTBRIDGE_DEVICE_ID) {
+    PciBase   = VMD_PCI_MMIO32_BASE;
+    PciSize   = VMD_PCI_MMIO32_SIZE;
+    PciIoBase = VMD_PCI_IO_BASE;
+    PciIoSize = VMD_PCI_IO_SIZE;
+    ASSERT (PlatformInfoHob->LowMemory <= PciBase);
+  } else if (PlatformInfoHob->HostBridgeDevId == INTEL_Q35_MCH_DEVICE_ID) {
+    PciBase = PlatformInfoHob->Uc32Base;
     //
     // The MMCONFIG area is expected to fall between the top of low RAM and
     // the base of the 32-bit PCI host aperture.
@@ -183,6 +189,7 @@ PlatformMemMapInitialization (
     ASSERT (PciExBarBase <= MAX_UINT32 - SIZE_256MB);
     PciSize = (UINT32)(PciExBarBase - PciBase);
   } else {
+    PciBase = PlatformInfoHob->Uc32Base;
     ASSERT (PlatformInfoHob->LowMemory <= PlatformInfoHob->Uc32Base);
     PciSize = 0xFC000000 - PciBase;
   }
@@ -364,6 +371,9 @@ PlatformMiscInitialization (
       break;
     case CLOUDHV_DEVICE_ID:
       break;
+    case VMD_HOSTBRIDGE_DEVICE_ID:
+      DEBUG ((DEBUG_INFO, "%a: OpenBSD vmd platform is configured.\n", __func__));
+      return;
     default:
       DEBUG ((
         DEBUG_ERROR,
